net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/exportCerts.php

```diff
@@ -51,19 +51,20 @@
     // lookup all config nodes
     if (isset($configObj->OPNsense->HAProxy->$key)) {
         foreach ($configObj->OPNsense->HAProxy->$key->children() as $child) {
-            // generate a crt-list for every child node
-            $crtlist = array();
-            $crtlist_filename = "/var/etc/haproxy/ssl/" . (string)$child->id . ".crtlist";
             // search in all matching child elements for ssl data
             foreach ($configNodes[$key] as $sslchild) {
                 if (isset($child->$sslchild)) {
-                    // multiple comma-separated values are possible
-                    $certs = explode(',', $child->$sslchild);
-                    foreach ($certs as $cert_refid) {
-                        // if the element has a cert attached, search for its contents
-                        if ($cert_refid != "") {
-                            // check all known cert types
-                            foreach ($certTypes as $type) {
+                    // generate a list for every known cert type
+                    foreach ($certTypes as $type) {
+                        // every child node needs its own set of lists
+                        $crtlist = array();
+                        $crtlist_filename = "/var/etc/haproxy/ssl/" . (string)$child->id . "." . $type . "list";
```

Comment:
pretty hard to trace the renamed / new files now, but don't know a better way either. anyway, it looks correct.

Reply:
@fichtner I was thinking about removing files that match the old name, what do you think? This cleanup task should then be removed after a few releases...

Reply:
sure, can do

in general we should however work towards a system that provides sensible information into a dedicated directory that is cleared on boot so that e.g. files that should be cleaned up regardless of installed / deinstalled plugin state are removed. that's usually why /tmp is useful here in that regard

Reply:
Sounds good, done in https://github.com/opnsense/plugins/pull/960/commits/47086ec1e34c0dc209a7c3a89f9cd875583381ea.